mtp_noms_ops/apps/security/views/check.py

```diff
@@ -66,16 +67,72 @@ def get_context_data(self, **kwargs):
         return context
 
 
-#  class AutoAcceptRuleDetailView(SecurityDetailView):
-    #  """
-    #  View history of all auto-accept rules
-    #  """
-    #  title = gettext_lazy('Auto Accept')
-    #  list_title = gettext_lazy('Auto Accepts')
-    #  #  template_name = 'security/auto_accept_rule.html'
-    #  object_context_key = 'auto_accept_rule'
-    #  id_kwarg_name = 'auto_accept_rule_id'
-    #  list_url = reverse_lazy('security:auto_accept_rule_list')
+class AutoAcceptRuleDetailView(SimpleSecurityDetailView, FormView):
+    """
+    View history of all auto-accept rules
+    """
+    list_title = gettext_lazy('Auto Accepts')
+    template_name = 'security/auto_accept_rule.html'
+    object_context_key = 'auto_accept_rule'
+    id_kwarg_name = 'auto_accept_rule_id'
+    list_url = reverse_lazy('security:auto_accept_rule_list')
+    form_class = AutoAcceptDetailForm
+
+    def get_form_kwargs(self):
+        return dict(super().get_form_kwargs(), request=self.request, object_id=self.kwargs[self.id_kwarg_name])
+
+    def get_object_request_params(self):
+        return {
+            'url': f'/security/checks/auto-accept/{self.kwargs[self.id_kwarg_name]}/'
+        }
+        return super().get_object_request_params()
+
+    def get_title_for_object(self, detail_object):
+        return '{} {} {} {}'.format(
+            gettext_lazy('Review auto accept of credits from'),
+            ','.join(detail_object['debit_card_sender_details']['cardholder_names']),
+            gettext_lazy('to'),
+            detail_object['prisoner_profile']['prisoner_name']
+        )
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+
+        if not self.object:
```

Comment:
it looks to me as thought the superclass already raises a 404 when the object can't be found. on the other hand, if it encounters an error loading the object, that's when it's an empty dict so really that should look like a 500 / "This service is currently unavailable" rather than a 404